elkscmd/basic/asm-ibmpc.S

```diff
@@ -0,0 +1,46 @@
+#define ARG0	2
+#define ARG1	4
+#define ARG2	6
+#define ARG3	8
+#define ARG4	10
+
+	.arch	i8086, nojumps
+	.code16
+	.text
+
+	.global fmemsetw
+	.global int_10
+
+// void fmemsetw(void * off, unsigned int seg, unsigned int val, size_t count)
+
+fmemsetw:
+	mov    %es,%bx
+	mov    %di,%dx
+	mov    %sp,%di
+	mov    ARG2(%di),%ax  // value
+	mov    ARG3(%di),%cx  // byte count
+	les    ARG0(%di),%di  // far pointer
+	cld
+	rep
+	stosw
+	mov    %dx,%di
+	mov    %bx,%es
+	ret
+
+// void int_10(unsigned int ax, unsigned int bx,
+//             unsigned int cx, unsigned int dx)
+
+int_10:
+	push   %bp
+	push   %si
+	push   %di
+	mov    %sp,%di
+	mov    ARG0+6(%di),%ax
+	mov    ARG1+6(%di),%bx
+	mov    ARG2+6(%di),%cx
+	mov    ARG3+6(%di),%dx
+	int    $0x10
```

Comment:
This ASM file is quite clean and nicely done, thank you. Much easier to read and understand than the C extensions, IMO and ready to be used by other programs more easily when needed thank you.

I was initially going to comment about using "ARG0+6" etc in this function, as well as the non-standard prologue and using DI to access the arguments, only to see that (my) fmemsetw function effective does the same thing! So I think everything is fine and will commit now.

Should the int_10 function be moved to a separate file in the future (leaving fmemset in its own fmemset.S file since it is written for very high speed w/non-standard prologue), then I would advise moving back to standard prologue and argument accessing, which will make things far easier if the function has to be used for medium model. This was not done for the original fmemsetw because that routine was specially written for the kernel and would always be accessed as a near procedure, even with some far functions, FYI.

So, when this function is moved into a separate file, or other functions are added, I would suggest using something like the following:
```
#define ARG0  4
#define ARG1  6
...
int_10:
 	push   %bp
        move   %sp,%bp   // Using BP ensures SS: used for access, unlike DI which requires DS==SS.
 	push   %si       // with BP=SP in previous step, any number of registers can be saved without adjusting ARGX
 	push   %di
 	mov    ARG0+FAR_ADJ(%bp),%ax // no +X offset difference per function, only change FAR_ADJ definition with small vs medium model
 	mov    ARG1+FAR_ADJ(%bp),%bx
        ...
```

Reply:
Hi @ghaerr 

Yes, assembling from C uses bp
but I referred to fmemsetw and used the di.
Let's fix it in the future. 

Thank you!